bin/add_frontend_server.rb

```diff
@@ -21,10 +66,21 @@
 
 instance_name = "frontend-#{determined_instance_zone[-1, 1] + (instance_count + 1).to_s}"
 
-puts "Naming instance #{instance_name}"
+puts "Naming instance #{instance_name}, verifying that the name is okay"
+
+Net::SSH.start('gateway.code.org', username, :password => password) do |ssh|
+  node_list = ssh.exec!("knife node list | egrep \'^#{instance_name}$\'")
+  throw 'Node name is currently in use. This should never happen - see if the host exists in EC2' unless node_list.nil?
+
+  client_list = ssh.exec!("knife client list | egrep \'^#{instance_name}$\'")
+  throw 'Client name is in use and is likely stale. Please log into gateway and delete the client, then rerun this command' unless client_list.nil?
```

Comment:
This line is a bit long especially given that the important 'unless' statement is at the end.

Could you wrap it and/or put the unless statement on a line of its own?